### PR TITLE
fix: auto migration column order unpredictable

### LINKED ISSUE #4351: Deterministic execution of AutoMigrate
## Describe the feature

At the moment, the order of which columns are migrated during `AutoMigrate` is non-deterministic. 

I'm using [copyist](https://github.com/cockroachdb/copyist) to test our migrations, which when run in "recording" mode, intercepts all SQL operations and saves them to a file. Then you can run the tests in "playback" mode and if the queries run do not match those which were recorded, then the test fails. This is great when you want to run a full test suite and you don't want to have a database running locally.

I'm creating a feature request, rather than a bug, since I don't think this would have been an intended feature unless someone else has specifically requested it.

## Motivation

Tests are good, tests without having to stand up a full database instance are even better.

Essentially, this for loop on line 99 iterates in a non-deterministic way, meaning sometimes my migrations migrate columns in one order, then in another run in another order: https://github.com/go-gorm/gorm/blob/2aca96d1474967da11bac81a58db9c97bd7bdcac/migrator/migrator.go#L96-L102

My request is to build a slice of column names, sort them, and then iterate over that slice instead.

## Related Issues

Couldn't find any related issues.

### PR DESCRIPTION
**Pull request by halfcrazy**
- [X] Do only one thing
- [X] Non breaking API changes
- [X] Tested

### What did this pull request do?
Fix #4351 make migration column order predictable.

### CODE REVIEW

File: tests/go.mod [hunk 1]
```diff
@@ -16,3 +16,5 @@ require (
 )
 
 replace gorm.io/gorm => ../
+
+replace gorm.io/driver/mysql => github.com/halfcrazy/mysql v1.2.3-0.20220106042630-805c93c0d8d9
```
Comment (jinzhu) [inline tests/go.mod:20]: Can you remove those debug code, thank you.

Comment (halfcrazy) [reply tests/go.mod]: done

Comment (halfcrazy) [reply tests/go.mod]: ci get failed after switching origin MySQL driver. pls take a look at https://github.com/go-gorm/mysql/pull/60

**Comment by jinzhu** [comment]
Hi @halfcrazy 

Can you format the source code with `gofmt` according to the bytedog suggestion? thank you.

**Comment by halfcrazy** [comment]
> Hi @halfcrazy
> 
> Can you format the source code with `gofmt` according to the bytedog suggestion? thank you.

Done

**Pull request merged**